offchain-modules/packages/x/src/handlers/ckb.ts

```diff
@@ -369,6 +410,128 @@ export class CkbHandler {
+        },
+      ]);
+      BridgeMetricSingleton.getInstance(this.role).addBridgeTxMetrics('ckb_lock', 'success');
+      BridgeMetricSingleton.getInstance(this.role).addBridgeTokenMetrics('ckb_lock', [
+        {
+          amount: Number(amount),
+          token: assetIdent,
+        },
+      ]);
+      logger.info(`CkbHandler watchLockEvents save CkbLock successful for ckb tx ${txInfo.tx.transaction.hash}.`);
+    }
+    if (records.length === 1) {
+      await this.db.updateLockConfirmNumber([{ ckbTxHash, confirmedNumber, confirmStatus }]);
+      logger.info(`update lock record ${txHash} status, confirmed number: ${confirmedNumber}, status: ${confirmed}`);
+    }
+    if (assetIdent == '0x0000000000000000000000000000000000000000000000000000000000000000') {
+      const ethereumMints = await this.db.getEthereumMintByCkbTxHashes([tx.hash]);
+      if (ethereumMints && ethereumMints.length === 1) {
+        const amountFromEthereumMint = BigInt(ethereumMints[0].amount);
+        await this.db.updateLockAmountAndBridgeFee([
+          {
+            ckbTxHash,
+            amount: `0x${amountFromEthereumMint.toString(16)}`,
+            bridgeFee: `0x${(amount - amountFromEthereumMint).toString(16)}`,
+          },
+        ]);
+        logger.info(
+          `get EthereumMint when check ckb lock ethereumMints for update ckb lock, records: ${records}, tx: ${tx}, bridgeFee: ${ethereumMints}`,
+        );
+      }
+    }
+
+    if (confirmed && this.role === 'collector') {
+      const ckbLocksSaved = await this.db.getCkbLockByTxHashes([ckbTxHash]);
+      if (ckbLocksSaved.length !== 1) {
+        logger.error(
+          `get ckb locks saved is not single: txHash: ${txHash}, ckbTxHash: ${ckbTxHash}, ckbLocksSaved: ${ckbLocksSaved}`,
+        );
+        return;
+      }
+      const ckbLock = ckbLocksSaved[0];
+      const filterReason = checkLock(amount, assetIdent, txHash, ckbLock);
+      if (filterReason !== '') {
+        logger.warn(`skip createEthereumMint for record: ${JSON.stringify(cellData)}, reason: ${filterReason}`);
+        return;
+      }
+      const nervosAsset = ForceBridgeCore.config.eth.nervosAssetWhiteList.find(
+        (asset) => asset.typescriptHash === assetIdent,
+      );
+      let mintAmount: bigint;
+      if (assetIdent == '0x0000000000000000000000000000000000000000000000000000000000000000') {
+        const bridgeFeeFromConfig = BigInt(ForceBridgeCore.config.eth.lockNervosAssetFee);
+        mintAmount = BigInt(ckbLock.amount) - bridgeFeeFromConfig;
+      } else {
+        mintAmount = BigInt(ckbLock.amount);
+      }
+
+      const erc20TokenAddress = nervosAsset!.xchainTokenAddress;
+      const mintRecords = [
+        {
+          ckbTxHash,
+          erc20TokenAddress,
+          nervosAssetId: assetIdent,
+          amount: `0x${mintAmount.toString(16)}`,
+          recipientAddress: recipientAddress,
+          blockNumber: blockNumber,
+          blockTimestamp: Number(block.header.timestamp),
```

Comment:
It's ethereum block, not nervos block, should be recorded by ethHandler not ckbHandler

Reply:
fixed, remove fields blockNumber and blockTimestamp and make them optional in IEthereumMint